configs,ext: gem5 SST bridge calls m5.instantiate() in gem5

kaustav-goswami opened the pull request:
This change updates the gem5 SST bridge to call m5.instantiate() in the gem5 config script instead of in the SST component. This allows more flexibility for the gem5-SST setup, as we can now write traffic generators using the bridge.

Change-Id: I510a8c15f8fb00bdbdd60dafa2d9f5ad011e48f2

kaustav-goswami committed:
configs,ext: gem5 SST bridge calls m5.instantiate() in gem5

This change updates the gem5 SST bridge to call m5.instantiate()
in the gem5 config script instead of in the SST component. This
allows more flexibility for the gem5-SST setup, as we can now write
traffic generators using the bridge.

Change-Id: I510a8c15f8fb00bdbdd60dafa2d9f5ad011e48f2
Signed-off-by: Kaustav Goswami <kggoswami@ucdavis.edu>

powerjg commented on ext/sst/gem5.hh:
```diff
@@ -80,7 +80,6 @@
 #include <sst/core/component.h>
 
 #include <sst/core/interfaces/stringEvent.h>
-#include <sst/core/interfaces/simpleMem.h>
```
Why this change?

kaustav-goswami replied:
My mistake here. This change was supposed to be in https://github.com/gem5/gem5/pull/396 to remove all deprecation warning messages. Do I revert this and create a different PR for this change? Or do I update the commit note?

powerjg replied:
You don't have to make a new PR, but if you could split this out into its on commit it would be better.

kaustav-goswami replied:
I have split this into two commits in the latest patch.

powerjg reviewed (commented):
Seems OK to me. One problem I foresee is that this isn't going to be compatible with how the stdlib does startup. This problem doesn't need to be solve now, but I would like to see the SST stuff be compatible with the stdlib in the long-term.

One small thing below to address.

kaustav-goswami commented:
> Seems OK to me. One problem I foresee is that this isn't going to be compatible with how the stdlib does startup. This problem doesn't need to be solve now, but I would like to see the SST stuff be compatible with the stdlib in the long-term.
> 
> One small thing below to address.

I agree that we cannot the simulator module with SST-based boards. I think to solve this issue, we need to make changes in the simulator module and add an (Mem) Interface that is based on the OutgoingRequestBridge.

kaustav-goswami committed:
ext: removed SST deprecation notice from SimpleMem

SST SimpleMem will be deprecated in SST 14. PR 396 updated the
bridge to use StandardMem, which is the new memory interface in
SST. This change removes all references to SimpleMem.

Change-Id: I6e4d645317d95ebb610e3dfc93a30d53b91b6b5d
Signed-off-by: Kaustav Goswami <kggoswami@ucdavis.edu>

powerjg commented:
Can you rebase on current develop?
